### PR TITLE
Add Just-in-time provisioning edit view

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
- Addresses https://github.com/wso2/identity-apps/issues/354.
- **This pull request depends on https://github.com/wso2/identity-apps/pull/715. Please start reviewing once it is merged. Otherwise, this PR contains both changes**.

## Goals
- $subject.

## Approach
- Supports configuring JIT provisioning.
Screenshot from 2020-04-18 02-23-15
Screenshot from 2020-04-18 02-23-22

### CODE REVIEW

File: apps/admin-portal/src/api/identity-provider.ts [hunk 1]
```diff
@@ -466,6 +467,41 @@ export const updateOutboundProvisioningConnector = (
         });
 };
 
+/**
+ * Update JIT provisioning configs of a specified IDP.
+ *
+ * @param idpId ID of the Identity Provider.
+ * @param configs JIT provisioning configs.
+ * @return {Promise<any>} A promise containing the response.
```
Comment (brionmario) [inline apps/admin-portal/src/api/identity-provider.ts:475]:
```suggestion
 * @return {Promise<IdentityProviderInterface>} A promise containing the response.
```

File: apps/admin-portal/src/api/identity-provider.ts [hunk 2]
```diff
@@ -466,6 +467,41 @@ export const updateOutboundProvisioningConnector = (
...
+ */
+export const updateJITProvisioningConfigs = (
+    idpId: string,
+    configs: JITProvisioningResponseInterface
+): Promise<any> => {
```
Comment (brionmario) [inline apps/admin-portal/src/api/identity-provider.ts:480]:
```suggestion
): Promise<IdentityProviderInterface> => {
```

File: apps/admin-portal/src/api/identity-provider.ts [hunk 3]
```diff
@@ -466,6 +467,41 @@ export const updateOutboundProvisioningConnector = (
...
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed to update identity provider: " + idpId));
+            }
+            return Promise.resolve(response.data as IdentityProviderInterface);
+        }).catch((error) => {
+            return Promise.reject(error);
```
Comment (brionmario) [inline apps/admin-portal/src/api/identity-provider.ts:501]: Shall we throw `IdentityAppsApiException` here.

File: apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    CommonPluggableComponentFormPropsInterface,
+    CommonPluggableComponentInterface,
+    CommonPluggableComponentMetaPropertyInterface,
+    CommonPluggableComponentPropertyInterface
+} from "../../../../models";
+import { Forms, FormValue } from "@wso2is/forms";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { getPropertyMetadata } from "../../utils";
+
+
```
Comment (vihanga-liyanage) [inline apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx:31]: I additional line

File: apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            } }
+            submitState={ triggerSubmit }
+        >
+            <Grid>
+                {
+                    dynamicValues && getSortedPropertyFields(metadata?.properties, false)
+                }
+                {
+                    enableSubmitButton && getSubmitButton("Update")
+                }
```
Comment (vihanga-liyanage) [inline apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx:190]:
```suggestion
 { dynamicValues && getSortedPropertyFields(metadata?.properties, false) }
 { enableSubmitButton && getSubmitButton("Update") }
```

File: apps/admin-portal/src/api/identity-provider.ts (hunk 3 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/api/identity-provider.ts]: Fixed in c0e366925209b7d80078ad6edc9077bb13ca9b5b

File: apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx (hunk 4 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/forms/components/common-pluggable-component-form.tsx:30]: Fixed in c0e366925209b7d80078ad6edc9077bb13ca9b5b

**Pull request merged**